src/components/fields/Image/SideDrawerField.tsx

```diff
@@ -84,57 +85,38 @@ export default function Image_({
   column,
   _rowy_ref,
   value,
-  onChange,
-  onSubmit,
   disabled,
 }: ISideDrawerFieldProps) {
   const confirm = useSetAtom(confirmDialogAtom, projectScope);
-  const { uploaderState, upload, deleteUpload } = useUploader();
-  const { progress } = uploaderState;
 
-  // Store a preview image locally while uploading
-  const [localImage, setLocalImage] = useState<string>("");
+  const { loading, progress, handleUpload, handleDelete, uploaderState } =
+    useFileUpload(_rowy_ref, column.key);
 
-  const onDrop = useCallback(
-    (acceptedFiles: File[]) => {
-      const imageFile = acceptedFiles[0];
+  const [localImages, setLocalImages] = useState<
+    (File & { localURL: string })[]
+  >([]);
 
-      if (_rowy_ref && imageFile) {
-        upload({
-          docRef: _rowy_ref! as any,
-          fieldName: column.key,
-          files: [imageFile],
-          previousValue: value ?? [],
-          onComplete: (newValue) => {
-            onChange(newValue);
-            onSubmit();
-            setLocalImage("");
-          },
-        });
-        setLocalImage(URL.createObjectURL(imageFile));
+  const { getRootProps, getInputProps, isDragActive } = useDropzone({
+    onDrop: async (acceptedFiles: File[]) => {
+      if (acceptedFiles.length > 0) {
+        setLocalImages(
+          acceptedFiles.map((file) =>
+            assignIn(file, { localURL: URL.createObjectURL(file) })
+          )
+        );
+        await handleUpload(acceptedFiles);
+        setLocalImages([]);
       }
     },
-    [_rowy_ref, value]
-  );
-
-  const handleDelete = (index: number) => {
-    const newValue = [...value];
-    const toBeDeleted = newValue.splice(index, 1);
-    toBeDeleted.length && deleteUpload(toBeDeleted[0]);
-    onChange(newValue);
-    onSubmit();
-  };
-
-  const { getRootProps, getInputProps, isDragActive } = useDropzone({
-    onDrop,
-    multiple: false,
+    multiple: true,
     accept: IMAGE_MIME_TYPES,
   });
```

Comment:
This code is repeated in File and Image for the Side Drawer and Table components, with very minor differences. It would be good to move it all into `useFileUpload`.

Reply:
[a3d2247](https://github.com/rowyio/rowy/pull/861/commits/a3d22474c9f6adbca39785c75ddc9224b691f2d8)